site/content/contribute/desktop/developer-setup.md

```diff
@@ -1,154 +1,75 @@
-
-## Tests
-Execute automated tests.
-
-```
-$ npm test
-```
-
-There are two steps in `npm test`.
-
-Test functionality:
-
-```
-$ npm run test:app
-```
-
-Test coding style:
-
-```
-$ npm run lint:js
-```
-
-## Helper commands
-
-### `npm run watch`
-Reload the application automatically when you have saved source codes.
-When using this mode, you can use "React Developer Tools" in the Developer Tools window.
-
-## Directory Structure
-```
-Mattermost Desktop
-├── docs/ - Documentations.
-├── resources/ - Resources which are used outside of the application codes, and original images of assets.
-├── scripts/ - Helper scripts.
-├── src/ - Application source code.
-│   ├── assets/ - Assets which are loaded from the application codes.
-│   ├── browser/ - Implementation of Electron's renderer process.
-│   │   ├── components/ - React.js components.
-│   │   ├── css/ - Stylesheets.
-│   │   ├── js/ - Helper JavaScript modules.
-│   │   └── webview/ - Injection code for Electron's <webview> tag.
-│   ├── common/ - Common JavaScript modules for both Electron's processes.
-│   └── main/ - Implementation of Electron's main process.
-│       └── menus/ - Application menu.
-└── test/ - Automated tests.
-    ├── modules/ - Scripts which are commonly used in tests.
-    └── specs/ - Test scripts.
-```
-
-### Other directories
-- `node_modules/` - Third party Node.js modules to develop and build the application.
-- `release/` - Packaged distributable applications.
-- `src/node_modules/` - Third party Node.js modules to use in the application.
+Set up your development environment for building, running, and testing the Mattermost Desktop App.
+
+## Dependencies
+
+<div class="tab">
+    <button class="tablinks active" onclick="openTab(event, 'mac')">macOS</button>
+    <button class="tablinks" onclick="openTab(event, 'windows')">Windows</button>
+    <button class="tablinks" onclick="openTab(event, 'ubuntu')">Ubuntu</button>
+    <button class="tablinks" onclick="openTab(event, 'archlinux')">Arch Linux</button>
+    <button class="tablinks" onclick="openTab(event, 'redhat')">Fedora/RedHat</button>
+</div>
+
+<div id="mac" class="tabcontent" style="display: block;">
+    {{% content "contribute/desktop/developer-setup/macos.md" %}}
+</div>
+
+<div id="ubuntu" class="tabcontent">
+    {{% content "contribute/desktop/developer-setup/ubuntu.md" %}}
+</div>
+
+<div id="windows" class="tabcontent">
+    {{% content "contribute/desktop/developer-setup/windows.md" %}}
+</div>
+
+<div id="archlinux" class="tabcontent">
+    {{% content "contribute/desktop/developer-setup/arch.md" %}}
+</div>
+
+<div id="redhat" class="tabcontent">
+    {{% content "contribute/desktop/developer-setup/redhat.md" %}}
+</div>
+
+#### Mattermost Server
+
+To develop with the Desktop App, you'll likely want to setup a Mattermost server and webapp instance that can be customized as needed, in case there are specific integration requirements needed for testing.
```

Comment:
```suggestion
To develop with the Desktop App, you'll likely want to set up a Mattermost server and webapp instance that can be customized as needed, in case there are specific integration requirements needed for testing.
```